src/deploy/deploy_to_pipedream.py

```
@@ -1317,5 +1402,73 @@ def main():
+        venv_python = venv_dir / "Scripts" / "python.exe"
+    else:
+        venv_python = venv_dir / "bin" / "python"
+
+    # Check if we're already running from venv (check path without resolving symlinks)
+    current_python = Path(sys.executable)
+    if venv_dir.exists() and str(venv_dir.resolve()) in str(current_python):
+        # Already in venv, just ensure playwright browser is installed
+        if not PLAYWRIGHT_AVAILABLE:
+            print("Installing Playwright browser...")
+            try:
+                subprocess.run([str(venv_python), "-m", "playwright", "install", "chromium"], check=True)
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)